x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/AmazonBedrockEmbeddingsRequestManager.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.sender;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.xpack.inference.common.Truncator;
+import org.elasticsearch.xpack.inference.external.http.retry.RequestSender;
+import org.elasticsearch.xpack.inference.external.request.amazonbedrock.embeddings.AmazonBedrockEmbeddingsEntityFactory;
+import org.elasticsearch.xpack.inference.external.request.amazonbedrock.embeddings.AmazonBedrockEmbeddingsRequest;
+import org.elasticsearch.xpack.inference.external.response.amazonbedrock.embeddings.AmazonBedrockEmbeddingsResponseHandler;
+import org.elasticsearch.xpack.inference.services.amazonbedrock.embeddings.AmazonBedrockEmbeddingsModel;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.inference.common.Truncator.truncate;
+
+public class AmazonBedrockEmbeddingsRequestManager extends AmazonBedrockRequestManager {
+    private static final Logger logger = LogManager.getLogger(AmazonBedrockEmbeddingsRequestManager.class);
+
+    private final AmazonBedrockEmbeddingsModel embeddingsModel;
+    private final Truncator truncator;
+
+    public AmazonBedrockEmbeddingsRequestManager(
+        AmazonBedrockEmbeddingsModel model,
+        Truncator truncator,
+        ThreadPool threadPool,
+        @Nullable TimeValue timeout
+    ) {
+        super(model, threadPool, timeout);
+        this.embeddingsModel = model;
+        this.truncator = Objects.requireNonNull(truncator);
+    }
+
+    @Override
+    public void execute(
+        String query,
+        List<String> input,
+        RequestSender requestSender,
+        Supplier<Boolean> hasRequestCompletedFunction,
+        ActionListener<InferenceServiceResults> listener
+    ) {
+        var serviceSettings = embeddingsModel.getServiceSettings();
+        var truncatedInput = truncate(input, serviceSettings.maxInputTokens());
+        var requestEntity = AmazonBedrockEmbeddingsEntityFactory.createEntity(embeddingsModel, truncatedInput);
+        var request = new AmazonBedrockEmbeddingsRequest(truncator, truncatedInput, embeddingsModel, requestEntity, timeout);
+        var responseHandler = new AmazonBedrockEmbeddingsResponseHandler();
+        var inferenceRequest = new ExecutableInferenceRequest(
```

Comment:
I think we might be able to bypass `ExecutableInterfaceRequest` and call `requestSender.send()` directly 🤷‍♂️ 

I'm also thinking we might not need to adhere to the `ResponseHandler` pattern here. It looks like that's only to pass the request through to the parsing logic.

What if we had `AmazonBedrockEmbeddingsRequest` instantiate a `AmazonBedrockEmbeddingsResponse` in the constructor. Then after executing the request it'd already have the response instance to pass the result to directly? I think this would avoid some of the casting and `isntanceof` check we do.

Reply:
We certainly could... I was trying to keep it like this to keep the same execution flow as everything else... I'll look into this a bit later.

Reply:
That would certainly simplify things... my thought was to keep everything as compatible with the existing paradigms though... but, if you're OK with shifting this, I'll take a look tomorrow and update it. Will certainly make it a lot cleaner than the async listener (for the AWS SDK result) - within - async listener (for the InferenceServiceResults) cocoon I have now...

Reply:
I think this morning, I see a bit more clear of what you're saying... and yeah I agree... we could go so far as to have the `execute` method here (and also for the chat completions) directly create and run a `AmazonBedrockExecutor` to perform the AWS call... the only downside I can see to this is the `requestSender` in the overridden `execute` method being ignored as we use a custom RequestSender for Amazon Bedrock as part of the sender service...

Reply:
... I was able to trim a bit of the fat around this, and also removed a couple of `instanceof` checks for creating the executor (by creating Amazon Bedrock executors specifically for chat completions and embeddings instead of the one generic base one...)... I think this is pretty solid at the moment